pyproject.toml

```diff
@@ -1,3 +1,7 @@
+[build-system]
```

Comment:
Is it necessary to add `build-system`?

Reply:
This is the classical way to package a python module, and documented in the `setuptools_scm` README: https://github.com/pypa/setuptools_scm#pyprojecttoml-usage.

Reply:
If I understand it correctly, your patch replaces `pkg_resources` and uses `importlib` to fetch the version.
So, the question is, do we need this section in `pyproject.toml` for fetching the version?
Or meybe I get it wrong

Reply:
This section is only about build: it follows python and `setuptools_scm` guidelines, and it makes the build more explicit about tools. I added it because it is in the README of `setuptools_scm`, which we are using to find the version of the package. In fact I could put this change in another commit or another PR if you want.

Reply:
I would put it in a separate commit.

Reply:
Done.